Assets/Scripts/PlayerSupervisor.cs

```diff
@@ -254,6 +265,92 @@ public int GetPoints()
         return points;
     }
 
+    /*
+    Variables:
+    1) # of blocks - either random # or set # you define above
+    2) min/max border values
+    3) random block lengths? true/false + min/max
+    4) random block heights? true/false + min/max
+    */
+    public void CreateTrainingBlocks()
+    {
+        if(multiBlockCreator.blocksPlacedRandomlyTF)
+        {
+            CreateRandomTrainingBlocks();
+        }
+        else{
+            CreateStaticTrainingBlocks();
+        }
+    }
+
+    public void CreateRandomTrainingBlocks()
+    {
+        int blockNum = 0;
+
+        if(multiBlockCreator.numBlocksChosenRandomlyTF)
+        {
+            blockNum = (int)Random.Range(5f, 50f);
+        }
+        else
+        {
+            blockNum = multiBlockCreator.numBlocksChoiceIfNotRandom;
+        }
+                    
+        for(int i = 0; i < blockNum; i++)
+        {
+            float randX = Random.Range(1f, 15f) + this.transform.parent.transform.position.x;
+            float randY = Random.Range(1f, 11f) + this.transform.parent.transform.position.y;
+
+            GameObject block = Instantiate(
+                blockGameObject, new Vector2(randX, randY),
+                Quaternion.identity, trainingBlocksGroup.transform
+            );
+            if(block)
+            {
+                block.GetComponent<Block>().playerSupervisor = this;
+                block.GetComponent<Block>().gameManager = this.GetComponent<PlayerSupervisor>().gameManager;
+            }
+                
+        }           
+    }
+
+    public void CreateStaticTrainingBlocks()
+    {
+        for(int i = 0; i < multiBlockCreator.numStaticBlocks; i++)
+        {
+            GameObject block = Instantiate(
+                blockGameObject, 
+                new Vector2(
+                    multiBlockCreator.staticBlockXPos[i] + this.transform.parent.transform.position.x,
+                    multiBlockCreator.staticBlockYPos[i] + this.transform.parent.transform.position.y
+                ), 
+                Quaternion.identity,
+                trainingBlocksGroup.transform
+            );
+
+            if(block)
+            {
+                block.GetComponent<Block>().playerSupervisor = this;
+                block.GetComponent<Block>().gameManager = this.GetComponent<PlayerSupervisor>().gameManager;
```

Comment:
There's only ever 1 GameManager instance, so the blocks don't need to worry about getting a specific instance.

Reply:
That makes sense. When I was setting this all up, not having the GM assignment in the code was causing it not to work but that could've been another item I addressed that made the GM assignment superfluous after it was fixed.